src/mars_patcher/minimap_tile_creator.py

```diff
@@ -0,0 +1,437 @@
+    ". - # . # . . .",
+    ". - # . # . . .",
+    ". . . . . . . .",
+    ". . . . . . . .",
+]
+PIXELS_MAP = [
+    ". . . . . . . .",
+    ". . . . . . . .",
+    ". # . . . # . .",
+    ". # # . # # . .",
+    ". # . # . # . .",
+    ". # . . . # . .",
+    ". . . . . . . .",
+    ". . . . . . . .",
+]
+
+PIXELS_BOSS = [
+    ". # . . # .",
+    ". # # # # .",
+    "# # # # # #",
+    "# . # # . #",
+    ". # # # # .",
+    ". # . . # .",
+]
+
+PIXELS_TUNNEL = [
+    "4 4 4 4 4 4 4 4",
+    "4 5 5 5 1 5 5 4",
+    "1 1 1 5 1 1 5 4",
+    "4 5 5 5 1 1 1 4",
+    "4 5 5 5 1 1 5 4",
+    "1 1 1 5 1 5 5 4",
+    "4 5 5 5 5 5 5 4",
+    "4 4 4 4 4 4 4 4",
+]
+
+PIXELS_MAJOR = [
+    ". . . . . . . .",
+    ". . . 6 6 . . .",
+    ". - 6 C C 6 - .",
+    ". 6 C 6 C C 6 .",
+    ". 6 C C C C 6 .",
+    ". - 6 C C 6 - .",
+    ". . . 6 6 . . .",
+    ". . . . . . . .",
+]
+PIXELS_CHOZO = [
+    ". . . . . . . .",
+    ". . 6 6 6 . . .",
+    ". 6 E E E 6 - .",
+    ". E E 6 E E 6 .",
+    ". E E E E E 6 .",
+    ". E 6 E E E 6 .",
+    ". 6 6 E E E 6 .",
+    ". . . . . . . .",
+]
+
+HAS_RED_OUTLINE = {
+    Content.EMPTY_RED_WALLS,
+    Content.NAVIGATION,
+    Content.SAVE,
+    Content.RECHARGE,
+    Content.HIDDEN_RECHARGE,
+    Content.DATA,
+    Content.SECURITY,
+}
+
+
+class TileSide(Enum):
+    TOP = 0
+    LEFT = 1
+    RIGHT = 2
+    BOTTOM = 3
+
+
+def create_tile(tile: MapTile) -> bytearray:
+    gfx = bytearray([(COLOR_BG << 4) | COLOR_BG for _ in range(32)])
+
+    # Handle tunnels separately
+    if tile.content == Content.TUNNEL:
+        draw_tunnel(gfx, tile.edges)
+        return gfx
+
+    # Corners
+    if tile.corners.top_left:
+        set_pixel(gfx, COLOR_WHITE_OUTLINE, 0, 0)
+    if tile.corners.top_right:
+        set_pixel(gfx, COLOR_WHITE_OUTLINE, 7, 0)
+    if tile.corners.bottom_left:
+        set_pixel(gfx, COLOR_WHITE_OUTLINE, 0, 7)
+    if tile.corners.bottom_right:
+        set_pixel(gfx, COLOR_WHITE_OUTLINE, 7, 7)
+
+    # Edges
+    for edge, side in zip(tile.edges, TileSide):
+        if isinstance(edge, Edge):
+            if edge == Edge.WALL:
+                draw_wall(gfx, side)
+            elif edge == Edge.DOOR:
+                draw_connection(gfx, side)
+        elif isinstance(edge, ColoredDoor):
+            if edge == ColoredDoor.BLUE:
+                draw_hatch_mf(gfx, COLOR_BLUE_EDGE, side)
+            elif edge == ColoredDoor.GREEN:
+                draw_hatch_mf(gfx, COLOR_GREEN_EDGE, side)
+            elif edge == ColoredDoor.YELLOW:
+                draw_hatch_mf(gfx, COLOR_YELLOW_EDGE, side)
+            elif edge == ColoredDoor.RED:
+                draw_hatch_mf(gfx, COLOR_RED_EDGE, side)
+            # TODO: ZM hatch types
+
+    # Content
+    # The following don't have doors near them so they aren't needed:
+    # Gunship, Animals, Boiler Pad
+    if tile.content in HAS_RED_OUTLINE:
+        draw_red_outline(gfx, tile.edges)
+
+    if tile.content == Content.NAVIGATION:
```

Comment:
instead of this weird if-else chain, i wonder whether it'd be better to have a dictionary that defines the data 
something along the lines of 
```py
{
Content.NAVIGATION: {
  draw_pixel_art: [COLOR_YELLOW_LETTER, PIXELS_NAVIGATION]
}
}
```
and then 
```py
if tile.content in my_dict:
  func, extra = next(my_dict[tile.content].items())
  func(gfx, *extra)
```
but dunno. seems harder to figure out at first glance on whats going on there?